consent-ws/src/main/java/org/broadinstitute/consent/http/service/DatabaseMatchingServiceAPI.java

```diff
@@ -105,15 +101,23 @@ public Match findMatchForPurpose(String purposeId){
     }
 
     private Match singleEntitiesMatch(Consent consent, Document dar) throws Exception {
+        if (consent == null) {
+            logger.error("Consent is null");
+            throw new IllegalArgumentException("Consent cannot be null");
+        }
+        if (dar == null) {
+            logger.error("Data Access Request is null");
+            throw new IllegalArgumentException("Data Access Request cannot be null");
+        }
         if(!dar.containsKey(DarConstants.RESTRICTION)){
-            logger().error("Error finding single matchData Access Request: "+ dar.getString(DarConstants.DAR_CODE) + " does not have a proper Use Restriction.");
+            logger.error("Error finding single matchData Access Request: "+ dar.getString(DarConstants.DAR_CODE) + " does not have a proper Use Restriction.");
             throw new Exception("Data Access Request: " + dar.getString(DarConstants.DAR_CODE) + " cannot be matched. Missing Use Restriction field.");
         }
-        if(consent != null && dar != null){
+        if (consent != null) {
```

Comment:
We already checked if consent was null above.

Reply:
Right ... I added the null check above to fail fast and neglected to clean up the if. Will fix!